### ISSUE TITLE
[v4] - Web bundle not building on Windows with metro plugin

### ISSUE THREAD

**Issue by rohanskalkat** [marklawlor/nativewind#784]
**Describe the bug**
When running the web app in expo using `yarn web` the build stops in the middle and metro server stops responding. This bug is only occurring on Windows, the same code builds successfully for web on macOS.

**To Reproduce**
1. Setup a new project with Expo SDK 50 + expo-router + nativewind v4 using `npx create-expo-stack`
2. Start web build using `yarn web`
3. **Error**: The build stops progressing randomly and the terminal stops responding

**Expected behavior**
The web build should finish and the project should load in the web browser.

**Expo Snack**
Here's a GitHub repo to reproduce the bug:
https://github.com/rohanskalkat/nativewind-metro-bug

**Screenshots**
![Screenshot 2024-02-09 222622](https://github.com/marklawlor/nativewind/assets/14858593/7a848020-88de-4905-a6a4-c8840feea9b0)

**Additional context**
 I have been debugging this and I have pinpointed the cause to `metro.config.js` file, if I remove the `withNativeWind` plugin from the file then the web build starts building successfully on Windows.

**Comment by SamuelLHuber** [comment]
would you mind sharing the working configuration ?  can't get it to work on linux also

**Comment by rohanskalkat** [comment]
Are you talking about env configuration? If so, here it is:

**Windows**
```
expo-env-info 1.2.0 environment info:
    System:
      OS: Windows 11 10.0.22631
    Binaries:
      Node: 20.11.0 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.19 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
      npm: 10.2.4 - C:\Program Files\nodejs\npm.CMD
    SDKs:
      Android SDK:
        API Levels: 28, 29, 30, 31, 32, 33, 33, 34
        Build Tools: 29.0.0, 29.0.2, 30.0.0, 30.0.2, 30.0.3, 31.0.0, 33.0.0, 33.0.1, 33.0.2, 34.0.0, 34.0.0, 34.0.0
        System Images: android-29 | Google APIs Intel x86 Atom, android-33 | Google APIs Intel x86_64 Atom, android-33 | Google Play Intel x86_64 Atom
    IDEs:
      Android Studio: AI-231.9392.1.2311.11330709
    npmPackages:
      expo: ^50.0.6 => 50.0.6 
      expo-router: ~3.4.7 => 3.4.7 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.73.4 => 0.73.4 
      react-native-web: ~0.19.6 => 0.19.10 
    Expo Workflow: bare
```

macOS

```
expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.3
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.16.0 - ~/.nvm/versions/node/v18.16.0/bin/node
      Yarn: 1.22.19 - /opt/homebrew/bin/yarn
      npm: 9.5.1 - ~/.nvm/versions/node/v18.16.0/bin/npm
      Watchman: 2023.12.04.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.12.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.0, iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0
    IDEs:
      Android Studio: 2022.3 AI-223.8836.35.2231.10811636
      Xcode: 15.0.1/15A507 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^50.0.6 => 50.0.6
      expo-router: ~3.4.7 => 3.4.7
      react: 18.2.0 => 18.2.0
      react-dom: 18.2.0 => 18.2.0
      react-native: 0.73.4 => 0.73.4
      react-native-web: ~0.19.6 => 0.19.10
    npmGlobalPackages:
      eas-cli: 5.9.3
    Expo Workflow: bare
```

**Comment by SamuelLHuber** [comment]
was more asking about the metro config

```
const { getDefaultConfig } = require('expo/metro-config');

/** @type {import('expo/metro-config').MetroConfig} */
// eslint-disable-next-line no-undef
const config = getDefaultConfig(__dirname);
```

still has the same issue you have for us on linux and on vercel builds

**Comment by rohanskalkat** [comment]
> was more asking about the metro config

Metro config is shown in the screenshot above and also available in the GitHub repo in the issue.

**Comment by marklawlor** [comment]
Please try `4.0.24`, it should fix bundling errors on windows. If not, please reply.

Additionally, any debugging you can provide would be greatly appreciated, the Metro config changes are fairly straight forward, even just working out which line it gets stuck on would greatly assist me.

**Comment by rohanskalkat** [comment]
Thanks @marklawlor the issue seems to be fixed, will update you if I see the issue again.

**Comment by dmitryshostak** [comment]
My team has just experienced the same issue when tried running our app on Windows 11 - the bundling gets stuck and unresponsive.

Reproducible on versions 4.0.33 - 4.0.36.

No errors and traces to debug. Please help 🙏

**Comment by dmitryshostak** [comment]
Hey @Knaackee ! Have you by any chance solved the issue?

**Comment by Knaackee** [comment]
> Hey @Knaackee ! Have you by any chance solved the issue?

The main problem is the lack of debug messages. 

So I stripped out everything and added back till I faced the error.

**Comment by dmitryshostak** [comment]
@Knaackee THIS HAS FIXED IT! https://github.com/marklawlor/nativewind/pull/8542

We had to change the node_modules file for now. I'll have to write the postinstall script for now 😢 

<img width="1255" alt="image" src="https://github.com/marklawlor/nativewind/assets/39539554/40c3c439-6444-4864-bd24-6090d3ccabe4">

@marklawlor please merge it

**Comment by feledori** [comment]
@dmitryshostak Just write a patch package for it. Yarn and pnpm support that and its easier than postinstall.

**Comment by prometixX** [comment]
I can confirm this issue is not solved in `4.0.36.`
The pull request mentioned by @dmitryshostak did resolve the stuck compilation on web.

**Comment by manssorr** [comment]
Check this patch: https://github.com/nativewind/nativewind/pull/860#issuecomment-2371839082
